### PR TITLE
Quality of Life Commit - File Library

### PR DESCRIPTION
**Pull request by ctomni231**
This commit introduces the File Library class, which is a dynamic loader
for server files. It also includes a file for displaying the FPS, and
does a small quality of life update for JSlix. This should cover Feature #457.

### CODE REVIEW

File: new_core/filelib.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	}
+
+	// This is queueNext, if ever needed
+	json.busy = 0;
+	if(json.queue.length > 0){
+		addFile(json.queue.shift());
+	}
+}
+
+//export function quickGet(text){
```
Comment (Tapsi) [inline new_core/filelib.js]: comments should be removed

File: new_core/fps.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  nowTime: 0,
+  lastTime: 0,
+  diffTime: 0
+}
+
+// default lx=4, ly=10
+export function display(ctx, lx, ly) {
+  time.nowTime = new Date().getTime();
+  time.diffTime = time.nowTime - time.lastTime;
+  time.tfps = parseInt(1000 / (time.diffTime || 1), 10);
```
Comment (Tapsi) [inline new_core/fps.js]: I still don't see the advantage of this one. Why don't extend the performance module by fps and tfps ? 

Like you can ask performance for the fps of the whole game loop and render it by the screen module. then only screen knows about rendering and performance about time calculation. this module knows both (or better is coupled to both terms)

this is something we should discuss at least. :)

File: new_core/fps.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  if(time.frame > 1000){
+     time.frame -= 1000;
+     if(time.frame > 1000){
+       time.frame = 0
+     }
+     time.jfps = time.count;
+     time.count = 0;
+  }
+
+  //ctx.fillStyle = '#FFFFFF';
```
Comment (Tapsi) [inline new_core/fps.js]: comments should be removed

File: new_core/filelib.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+		return;
+	}
+
+	json.busy = 1
+
+	// This will get the right object for XMLHttpRequest
+	if(window.XMLHttpRequest){
+		alert("Gets here")
+		json.request = new XMLHttpRequest();
+	}else if(window.ActiveXObject){
```
Comment (Tapsi) [inline new_core/filelib.js]: we have no IE support, so we should get rid of any ActiveXObject mechanic

File: new_core/filelib.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+	if(json.busy == 1){
+		json.queue.push(fileurl);
+		return;
+	}
+
+	json.busy = 1
+
+	// This will get the right object for XMLHttpRequest
+	if(window.XMLHttpRequest){
+		alert("Gets here")
```
Comment (Tapsi) [inline new_core/filelib.js]: can you replace this by console.log?

File: new_core/json/options.json [hunk 6]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (Tapsi) [inline new_core/json/options.json]: file should be removed because options is included in state.js (see #454)

File: new_core/screens/initial.js [hunk 7]
```diff
@@ -40,18 +49,24 @@ export function update () {
...
   ctx.clearRect(0, 0, canvas.width, canvas.height);
 
+  ctx.fillStyle = "rgb(" + red + "," + green + "," + blue + ")"
   ctx.fillRect(0,0,canvas.width,canvas.height)
 
   // This draws the image onto the screen
 	ctx.drawImage(jslix.getImg(0), 90, 120, 200, 20)
   //ctx.drawImage(jslix.quickImage("../image/mobile/cwttitle.png"), 10, 120, 200, 20)
 
+  fps.display(ctx, 4, 10)
```
Comment (Tapsi) [inline new_core/screens/initial.js]: when we move this line into screen.js after line 19, then all screens can show the fps :)

File: new_core/screens/initial.js [hunk 8]
```diff
@@ -40,18 +49,24 @@ export function update () {
   green = parseInt(colorTween.green.value, 10)
   blue = parseInt(colorTween.blue.value, 10)
 
+  // This is to finish testing the dynamic Json functionality
```
Comment (Tapsi) [inline new_core/screens/initial.js]: no code-comments in master please

File: new_core/fps.js (hunk 2 above)
Comment (Tapsi) [reply new_core/fps.js]: we solving this in a new issue

File: new_core/filelib.js (hunk 5 above)
Comment (ctomni231) [reply new_core/filelib.js]: Yeah, I think a bit of cleaning up alerts is in order here

File: new_core/filelib.js (hunk 4 above)
Comment (ctomni231) [reply new_core/filelib.js]: Yeah, I also agree with this change. Internet Explorer, a relic from an older time.

File: new_core/screens/initial.js (hunk 7 above)
Comment (ctomni231) [reply new_core/screens/initial.js]: Yeah, that is a very good move. I think merging this with the performance.js file would be a decent move, but I think just closing these conversations and starting fresh would be a good move considering the high amount of changes to get this in line.

**Comment by ctomni231** [comment]
Okay, pushed a few more changes into the featurejson branch. Hopefully, this will clear up a lot of the lingering bugs we've discussed. Awaiting your review :)

**Pull request merged**